test/test_torch.py

```diff
@@ -4486,6 +4486,24 @@ def test_accreal_type(self):
         self.assertIsInstance(x.char().sum(), int)
         self.assertIsInstance(x.byte().sum(), int)
 
+    def test_new(self):
+        x = torch.autograd.Variable(torch.Tensor())
+        y = torch.autograd.Variable(torch.randn(4, 4))
+        z = torch.autograd.Variable(torch.IntTensor([1, 2, 3]))
+        self.assertEqual(x.new(), x)
```

Comment:
It is the same semantics as `Tensor.new`. `x.new()` returns a new empty tensor.

We've been returning `[0]` for the shape since https://github.com/pytorch/pytorch/commit/067f799e9ff3cc24010492d512ad610caf4143ea (three weeks ago).